packages/@textlint/types/src/Plugin/TextlintPluginModule.ts

```diff
@@ -40,7 +40,10 @@ export declare class TextlintPluginProcessor {
          * @param text
          * @param filePath
          */
-        preProcess(text: string, filePath?: string): TxtNode | { text: string; ast: TxtNode };
+        preProcess(
+            text: string,
+            filePath?: string
+        ): TxtNode | { text: string; ast: TxtNode } | Promise<TxtNode> | Promise<{ text: string; ast: TxtNode }>;
```

Comment:
Maybe,

- Add `PreProcessResullt` type
- Return `PreProcessResullt | Promise<PreProcessResullt>`

It is more reasonable.

```ts
type PreProcessResult = TxtNode | { text: string; ast: TxtNode };
...
        preProcess(
            text: string,
            filePath?: string
        ): PreProcessResullt | Promise<PreProcessResullt>;
```